### DISCUSSION TITLE
请问如何在非叶子节点的位置插入自定义节点

### DISCUSSION THREAD

**Discussion by liran19960215** [antvis/S2#2240] [Q&A]
目前官网上的提供了自定义的demo，如下图所示，可以将原来的“宁波市”替换成“宁波A”和“宁波B”，或者可以在改层级插入其他节点，https://s2.antv.antgroup.com/examples/custom/custom-layout/#custom-layout-hierarchy
![image](https://github.com/antvis/S2/assets/65275407/6895499c-0700-4863-8000-0a12181f04f4)
但是想请问下如何在有子节点的层级插入自定义节点，如下图所示，即在“四川省”下面插入“安徽省”类似的功能
![image](https://github.com/antvis/S2/assets/65275407/8750dc1f-ae19-4130-94bd-2b73c685229a)
这种目前有办法支持吗？

谢谢！

**Comment by wjgogogo** [comment]
那为什么不在传入数据的时候就将安徽省传进去呢？具体说一下使用场景

**Reply by liran19960215** [reply]
目前的使用场景是想插入迷你图列，应该是列最外层保持同级的，例如当迷你图by指标的时候（两列），列头分别是”迷你图“ ，然后其叶子节点对应各”指标名“，但是现在无法满足这种情况。还有一种使用场景是行列计算，如果是列上的排列是维度+指标，选中维度应该也是可以进行行列计算的，但是现在目前只能在叶子节点插入自定义节点，所以这种情况下只能够选中指标进行行列计算。

**Reply by liran19960215** [reply]
![image](https://github.com/antvis/S2/assets/65275407/9e4140be-2a4b-4e69-a52d-ff7e491f54a0)
如图所示，在多度量值的情况下希望能够插入迷你图列，然后通过改写getCellMeta，再渲染到迷你图时再动态的获取到对应行所有的值，这样就不能够只在叶子节点插入新节点，列头需要插入两行节点。
